src/app/products_view/product_availability_details/product_availability_details.ts

```diff
@@ -0,0 +1,21 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { ProductQuantityInfo } from '../../shared/model/product_quantity_info';
+
+@Component({
+  selector: 'tk-product-availability-details',
+  templateUrl: './product_availability_details.ng.html',
+  styleUrls: ['./product_availability_details.scss'],
+})
+export class ProductAvailabilityDetails implements OnInit {
+  @Input() productQuantityInfo: ProductQuantityInfo[] = [];
+
+  selectedProductQuantityInfo: ProductQuantityInfo;
+
+  ngOnInit() {
+    this.selectedProductQuantityInfo = this.productQuantityInfo[0];
```

Comment:
This will throw an error
`ERROR TypeError: Cannot read properties of undefined (reading '0')`
when the product info array is empty, please make sure we handle this case appropriately.